### DISCUSSION TITLE
Document versioning

### DISCUSSION THREAD

**Discussion by Ekajto** [docToolchain/docToolchain#647] [Q&A]
Hello

I am wondering how to use docToolchain to help me with versioning of documentation. 
I have following use case:
There is a release of a product and along with it we need to release a documentation that is compatible with product in specific version, after some time product is released with new version and we need also new version of documentation but old one need to be kept for users that are using older version of product. 
Everything needs to be stored in some way on confluence(pages, pdf) and be automated through jenkins. 

I have started to try to inject some custom attributes when running `./dtcw publishToConfluence -Pversiondoc=2` in docToolchainConfig.groovy I added following line pageSuffix = ' - ' + ${versiondoc} in publishToConfluence configuration but no luck. Pages were created with pageName - null.

Is such injection possible? How are you dealing with documentation versioning? What are your practices with automated process through jenkins or other tools? 

Best Regards

**Comment by rdmueller** [comment]
such an injection should be possible. 
take a look at the `docToolchainConfig.groovy`! It is a groovy file which gets execute.

you could specify your pageSuffix as

```groovy
pageSuffix = System.getenv('versiondoc') ?: ''
```

this way, it will use the environment variable `vresiondoc` as pageSuffix or the empty String `` if the environment variable is not set.

Does this work for you?

**Reply by Ekajto** [reply]
Ok but how can I pass this variable value when using docker image (as this seems as the most convenient way as everything is setup). Is there a way to do it without doing some manual changes in wrapper script? 
`./dtcw publishToConfluence -Pversiondoc=2` does not work if I setup environment variable in jenkins it will also not work as internal docker is launched which performs given task so it is required to pass such environment variable inside docker to make it work. 

Or am I missing something?

**Reply by rdmueller** [reply]
Hi! Here is the solution:

use
```
   example = System.properties.myProperty
```
in your config. This property can be set via the command line

```
./dtcw docker generateHTML -DmyProperty=myValue
```

I've also added this to the manual, but it will take a minute to show up:
https://doctoolchain.github.io/docToolchain/v2.0.x/010_manual/100_config.html

**Comment by rdmueller** [comment]
another solution could be to store older versions only as PDF. exportToConfluence can automatically upload such a PDF...

**Comment by rdmueller** [comment]
Good Morning! did one of the solutions work for you?
